### PR TITLE
Add vimeo/psalm as new tool

### PR DESCRIPTION
**Pull request by MacFJA**
PR for the issue #85

### CODE REVIEW

File: app/psalm.xml.twig [hunk 1]
```diff
@@ -0,0 +1,40 @@
+<?xml version="1.0"?>
```
Comment (zdenekdrahos) [inline app/psalm.xml.twig]: Great idea to define the XML template in twig!

File: composer.lock [hunk 2]
```diff
@@ -4,8 +4,7 @@
         "Read more about it at https://getcomposer.org/doc/01-basic-usage.md#composer-lock-the-lock-file",
```
Comment (zdenekdrahos) [inline composer.lock]: Please revert changes in `composer.lock`.
I know that lock file is not up to date. I'll update before releasing a new version.

File: src/CodeAnalysisTasks.php [hunk 3]
```diff
@@ -85,6 +85,12 @@
             'outputMode' => OutputMode::RAW_CONSOLE_OUTPUT,
             'composer' => 'phpunit/phpunit',
         ),
+        'psalm' => array(
+            'optionSeparator' => '=',
+            'xml' => ['psalm.xml'],
+            'errorsXPath' => '//item/severity[text()=\'error\']',
```
Comment (zdenekdrahos) [inline src/CodeAnalysisTasks.php]: Add [`internalClass`](https://github.com/EdgedesignCZ/phpqa/blob/master/src/CodeAnalysisTasks.php#L72). It prevents running [a suggested tool](https://github.com/EdgedesignCZ/phpqa#suggested-tools)  when the tool is not installed. 
phpqa should quietly ignore the tool, not show an error message:

```bash
# bad
$ ./phpqa --verbose --report --config tests/.travis --tools psalm
 [FileSystem\CleanDir] Cleaned build/
 [FileSystem\FilesystemStack] mkdir ["build\/"]
 [Edge\QA\Task\ParallelExec] "/phpqa-psalm/vendor/bin/psalm" --config="build/psalm-config.xml" --output-format=console --show-info=true --report="build//psalm.xml"
 1/1 [============================] 100%
 [Edge\QA\Task\ParallelExec] Output for  "/phpqa-psalm/vendor/bin/psalm" --config="build/psalm-config.xml" --output-format=console --show-info=true --report="build//psalm.xml" 

sh: 1: /phpqa-psalm/vendor/bin/psalm: not found

# good
$ ./phpqa --verbose --report --config tests/.travis --tools psalm
 [FileSystem\CleanDir] Cleaned build/
 [FileSystem\FilesystemStack] mkdir ["build\/"]
    0 [>---------------------------]
 [Edge\QA\Task\ParallelExec] 0 processes finished running
 [Edge\QA\Task\ParallelExec] Done in 0s
```


After that you can add psalm to [bin/ci.sh](https://github.com/EdgedesignCZ/phpqa/commit/41c3ebeb264dd0ebeb685d81ee7e7e46203efe58#diff-2a1b94d93450647d2b30d54be117512fR3) (_it will be tested in [php 7.1](https://github.com/EdgedesignCZ/phpqa/blob/master/.travis.yml#L19)_)

File: src/CodeAnalysisTasks.php [hunk 4]
```diff
@@ -463,6 +469,44 @@ private function phpunit(RunningTool $tool)
...
+                $twig->render(
+                    'psalm.xml.twig',
+                    array(
+                        'includes' => $this->options->getAnalyzedDirs(),
+                        'excludes' => $this->options->ignore->psalm()
+                    )
+                )
+            );
+            $psalmConfigFile = escapePath($psalmConfigFile);
+        }
```
Comment (zdenekdrahos) [inline src/CodeAnalysisTasks.php]: Add [documentation](https://github.com/EdgedesignCZ/phpqa/commit/a4bf0ca44b4bb1e474113d5fe6b95de984ce33c8#diff-04c6e90faac2675aa89e2176d2eec7d8) ([#suggested tools](https://github.com/EdgedesignCZ/phpqa#suggested-tools), [#output modes](https://github.com/EdgedesignCZ/phpqa#output-modes), [#advanced configuration](https://github.com/EdgedesignCZ/phpqa#advanced-configuration---phpqayml)).
Please mention in `psalm.config` that `analyzedDirs` and `ignoredDirs` are not used when custom config is specified (_you can copy-paste it from `phpunit.config`_).

**Comment by MacFJA** [comment]
@zdenekdrahos Is it good or you need more?

**Comment by zdenekdrahos** [comment]
Perfect!
Thanks, @MacFJA.

**Pull request merged**